ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -71,44 +74,91 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun WearApp(watchRepository: WatchRepository) {
     WearAppTheme {
-        // The state for the ScalingLazyColumn in other screen is hoisted to this level, so
-        // the Scaffold can properly place the position indicator (also known as the
-        // scroll indicator). We use it for various other things (like hiding the time when
-        // the user is scrolling).
-        //
-        // This current solution works for one scrolling type Composable, that is, we only
-        // have one Composable capable of scrolling.
-        val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+        // The state for any scrollable content in the composable screens is hoisted to this level,
+        // so the Scaffold can properly place the position indicator (also known as the
+        // scroll indicator).
+        // In our case, the watch list screen uses a ScalingLazyColumn and the watch details
+        // screen uses a Column with scrolling enabled.
+        // We also use the states for various other things (like hiding the time when the user is
+        // scrolling).
+        // ----------
+
+        // For all scrolling screens, we save their scrolling offset in a remembered map (as a
+        // ScrollableState), so when users return to the screen, they are at the same location as
+        // they left.
+        // We set the default to the watch list screen (ScalingLazyListState) since there is only
+        // one. For the watch detail screens (with many different variations based on the watch id),
+        // we add those dynamically to the map as they are needed (in the navigation section).
+        var currentScrollStateKey: String = remember { Screen.WatchList.route }
```

Comment:
This line seems very suspicious to me.

There isn't a `mutableStateOf` here, so any time `currentScrollStateKey` is set, it will just be wiped out with next recomposition.